core/src/test/scala/com/stripe/dagon/RealNumberTest.scala

```diff
@@ -193,12 +199,14 @@ object RealNumbers {
      */
     def divOpt(r0: Real): Option[Real] =
       r0 match {
+        case z if z.isDefinitelyZero => None
+        case c@Const(_) if !c.isFinite => None
         case same if same == self => Some(one)
         case Prod(ps) =>
           // to divide by a product all must divide
           @annotation.tailrec
           def loop(r: Real, ps: SortedList[Real]): Option[Real] =
-            if (ps.isEmpty) Some(one)
+            if (ps.isEmpty) Some(r)
```

Comment:
it is easy to get it backwards what we are doing, here `Prod(SortedList.empty) == 1` but `r/1 = r` not `1`. This was a key error.